packages/rocket-batch-file-process-aws-infrastructure/test/unit/batch-file-processing.test.ts

```diff
@@ -0,0 +1,163 @@
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { App, IConstruct, Stack, StackProps } from '@aws-cdk/core'
+import { Function, Runtime, Code } from '@aws-cdk/aws-lambda'
+
+import { BatchFileProcessingStack, AWSBatchProcessingFilesParams } from '../../src/batch-file-processing'
+import { Bucket, EventType } from '@aws-cdk/aws-s3'
+import { RocketUtils } from '@boostercloud/framework-provider-aws-infrastructure'
+
+import * as chai from 'chai'
+import * as chaiSubset from 'chai-subset'
+
+import { AttributeType, Table } from '@aws-cdk/aws-dynamodb'
+import { fake } from 'sinon'
+
+chai.use(require('sinon-chai'))
+chai.use(require('chai-as-promised'))
+chai.use(chaiSubset)
+
+const expect = chai.expect
+
+describe('Batch file processing stack', () => {
+  let appStack: Stack
+  let bucketName: string
+  let config: BoosterConfig
+  let chunkSize: string
+  let environmentToCheck: any
+  let params: AWSBatchProcessingFilesParams
+  let eventsTable: Table
+  let eventHandlerFunction: Function
+
+  before(() => {
+    bucketName = 'sample-bucket'
+    config = new BoosterConfig('test')
+    chunkSize = '10'
+    config.appName = 'testing-app'
+
+    appStack = new Stack(new App(), config.resourceNames.applicationStack, {} as StackProps)
+
+    // Added dummy infra
+    const eventStoreName = 'events-store'
+    const eventsMain = 'events-main'
+    eventsTable = new Table(appStack, eventStoreName, {
+      tableName: eventStoreName,
+      partitionKey: { name: 'id', type: AttributeType.STRING },
+    })
+
+    eventHandlerFunction = new Function(appStack, eventsMain, {
+      memorySize: 1024,
+      handler: 'index.handler',
+      runtime: Runtime.NODEJS_12_X,
+      code: Code.fromInline('{}'),
+    })
+
+    environmentToCheck = {
+      EVENT_STORE_NAME: { value: eventsTable.tableName },
+      CHUNK_SIZE: { value: chunkSize },
+      STAGING_BUCKET_NAME: { value: `${bucketName}-staging` },
+      ENTITY_TYPE_NAME: { value: 'File' },
+      TYPE_NAME: { value: 'FileAdded' },
+    }
+
+    params = {
+      bucketName,
+      chunkSize,
+    }
+  })
+
+  context('on mount', () => {
+    const verifyFunctionPolicies = (currentPolicies: any[], expectedPolicies: any[]) => {
+      expectedPolicies.forEach((item) => {
+        const plcy = currentPolicies.find((policy) => {
```

Comment:
I think it'd be better inverting the names:
```suggestion
        const policy = currentPolicies.find((plcy) => {
```